src/scripts/util/data.model.js

```diff
@@ -79,6 +79,39 @@ function valueDomain(model) {
     }
     return domain;
 }
+*/
+
+function valueDomain(model){
+    if(model.valueDomain){return model.valueDomain;}
+
+    var domain, extents;
+    if(model.type !== 'stacked'){
+        var extents = setExtents(model);
+        var domain = d3.extent(extents);
+        extents = setExtents(model);
+        domain = d3.extent(extents);
+
+        if(!model.falseOrigin && domain[0] > 0){
+            domain[0] = 0; //see line 88
+        }
+    }else{
+        var vals = [];
+
+        model.data.map(function (d, j){
+        var k, sum = 0;
+        for(k in d.values[0]){
+        if(k !== 'date'){
+            sum += d.values[0][k];
+        }
+    }
+    vals.push(sum);
+});
+
+domain = d3.extent(vals);
+domain[0] > 0 ? domain[0] = 0 : 0; //if we do a chart where we don't want the start of the Y-axis to be 0, we'll want to be able to specify a lowest value other than 0
+}
+return domain;
+}
```

Comment:
When trying to understand this i noticed the hard-coded `date` value. should this be `model.x.series.key`?

I had a go at refactoring it to try and answer my next question around what the for loop with the conditional statement is trying to achieve.  Does the below make sense? i just pulled the loop+if nesting into its own function to help with self-documentation.

```javascript
function sumStackedValues(model){
    var extents = [];
    model.data.map(function (d, j) {
        var key, sum = 0;
        for (key in d.values[0]) {
            if (key !== 'date') {
                sum += d.values[0][key];
            }
        }
        extents.push(sum);
    });
    return extents;
}

function valueDomain(model){
    if(model.valueDomain){ return model.valueDomain; }

    var extents = (model.type === 'stacked') ? sumStackedValues(model) : setExtents(model);
    var domain = d3.extent(extents);
    if(!model.falseOrigin && domain[0] > 0){
        domain[0] = 0;
    }
    return domain;
}
```